rosbag2_cpp/include/rosbag2_cpp/writer_interfaces/base_writer_interface.hpp

```diff
@@ -45,6 +45,8 @@ class ROSBAG2_CPP_PUBLIC BaseWriterInterface
   virtual void remove_topic(const rosbag2_storage::TopicMetadata & topic_with_type) = 0;
 
   virtual void write(std::shared_ptr<rosbag2_storage::SerializedBagMessage> message) = 0;
+
+  virtual bool request_owned_serialized_data() = 0;
```

Comment:
Non-blocking: This feels very much like an under-the-hood implementation detail. It would be nice if we could avoid exposing it to the global interface. However - I don't know if we can do that right now. We may just need to wait until we remove the `write(SerializedBagMessage &)` interface.

Reply:
Currently, I not find better way instead of adding hook function to global interface.